ckan/logic/validators.py

```diff
@@ -233,6 +232,33 @@ def resource_id_does_not_exist(
     if parent_id != package_id:
         errors[key].append(_('Resource id already exists.'))
 
+def id_does_not_exist(value: str, context: Context) -> Any:
+    """Ensures that the value is not used as a ID or name.
+    """
+
+    model = context['model']
+    session = context['session']
+    entities_map = {
+        'resource_view': model.ResourceView,
+        'group': model.Group
```

Comment:
@shashigharti I think in this case it's safer to have individual validators for each entity type even if that means many different functions. Relying on the keys to determine the entity type won't always be safe. And it will make overriding specific validators easier for plugins.
So let's go for `resource_view_id_does_not_exist`, `group_id_does_not_exist`, etc